### DISCUSSION TITLE
conditionally re-fetching data with SWR

### DISCUSSION THREAD

**Discussion by Abdelmuttalib** [vercel/swr#1648] [Q&A] [answered]
Is there a way I can re-fetch data and make the request all over again on a certain condition with SWR? 
I have a page where I'm displaying a list of items and there is a dropdown on the page to allow users to filter results, depending on what the user chooses to be ordered by, I have to re-fetch data with the needed query parameters. Is there a way I can re-fetch at the certain conditions when the user specifies a filter for example 'best-selling'.

**Comment by shuding** [comment, marked as answer]
When the SWR key changes, it will refetch automatically. So you can put the key in a state and update it when needed.
